dace/autodiff/implementations/onnx_ops.py

```diff
@@ -0,0 +1,1165 @@
+                with dace.tasklet:
+                    v >> maxv
+                    v = -9999999
+
+                # Deterministic argmax (assuming sequential map)
+                for i, j in dace.map[0:sy, 0:sx]:
+                    with dace.tasklet:
+                        o << X[b, c, sty * ti + i, stx * tj + j]
+                        vin << maxv
+                        v >> maxv(-1)
+                        ind_i >> maxi(-1)
+                        ind_j >> maxj(-1)
+                        if o > vin:
+                            v = o
+                            ind_i = i
+                            ind_j = j
+                with dace.tasklet:
+                    igrad << Y_grad[b, c, ti, tj]
+                    ind_i << maxi
+                    ind_j << maxj
+                    ograd >> X_grad(1)[b, c, :, :]
+                    ograd[ind_i, ind_j] = igrad
+
+        result_node, result = butils.backward_program_for_node(maxpool_backward, context, forward_node)
+
+        _find_map_by_param(result_node.sdfg, 'i').schedule = \
+            dace.ScheduleType.Sequential
+
+        return result_node, result
+
+
+@autoregister_params(op="LogSoftmax", name="default")
+class DefaultLogSoftmaxBackward(BackwardImplementation):
+    """Backward implementation for ONNX LogSoftmax operation.
+
+    Computes gradients using the mathematical relationship for log-softmax:
+    dX = dY - exp(Y) * sum(dY)
+    where Y is the forward output and dY is the output gradient.
+    """
+
+    @staticmethod
+    def backward(forward_node: nd.Node, context: BackwardContext, given_gradients: List[Optional[str]],
+                 required_gradients: List[Optional[str]]) -> Tuple[nd.Node, BackwardResult]:
+
+        dim = forward_node.axis
+        output_shape = butils.forward_out_desc_with_name(forward_node, context, "output").shape
+        output_dtype = butils.forward_out_desc_with_name(forward_node, context, "output").dtype
+
+        sums_shape = list(copy.deepcopy(output_shape))
+        sums_shape[dim] = 1
+
+        def logsoftmax_backward(output, output_grad, input_grad):
+            exp_output = dace.define_local(output_shape, output_dtype)
+            donnx.ONNXExp(input=output, output=exp_output)
+
+            grad_output_sum = dace.define_local(sums_shape, output_dtype)
+            donnx.ONNXReduceSum(data=output_grad, reduced=grad_output_sum, keepdims=1, axes=[dim])
+            # let's not use ONNXMul here; not sure how this inplace op is handled by ORT...
+            exp_output[:] = exp_output * grad_output_sum
+            donnx.ONNXSub(A=output_grad, B=exp_output, C=input_grad)
+
+        result_node, result = butils.backward_program_for_node(logsoftmax_backward, context, forward_node)
+
+        butils.connect_output_from_forward(forward_node, result_node, context, "output")
+        return result_node, result
+
+
+@autoregister_params(op="Conv", name="PyTorch-dwise")
+class PyTorchConvBackward(BackwardImplementation):
```

Comment:
I'd move this one to a different file. This is a bit more of a pytorch-specific node